packages/spacecat-shared-data-access/src/models/importer/import-job.js

```diff
@@ -44,127 +40,129 @@ const ImportJob = (data) => {
   self.getInitiatedBy = () => self.state.initiatedBy;
 
   /**
-     * Updates the end time of the ImportJob.
-     * @param {string} endTime - The new end time.
-     * @returns {ImportJob} The updated ImportJob object.
-     */
-  self.updateEndTime = (endTime) => {
-    if (!isIsoDate(endTime)) {
-      throw new Error(`Invalid end time during update: ${endTime}`);
+   * Updates the state of the ImportJob.
+   * @param key - The key to update.
+   * @param value - The new value.
+   * @param validator - An optional validation function to use before updating the value.
+   * @returns {ImportJob} The updated ImportJob object.
+   */
+  const updateState = (key, value, validator) => {
+    if (validator && typeof validator === 'function') {
+      validator(value);
     }
 
-    self.state.endTime = endTime;
+    self.state[key] = value;
     self.touch();
 
     return self;
   };
 
   /**
-     * Updates the duration of the ImportJob.
-     * @param {number} duration - The new duration.
-     * @returns {ImportJob} The updated ImportJob object.
-     */
-  self.updateDuration = (duration) => {
-    if (!isNumber(duration)) {
-      throw new Error(`Invalid duration during update: ${duration}`);
+   * Updates the end time of the ImportJob.
+   * @param {string} endTime - The new end time.
```

Comment:
Worth mentioning the format it needs to be in?

Reply:
Updated.